### ISSUE TITLE
NERvCopyString might return an incorrect value in some case

### ISSUE THREAD

**Issue by RangerCD** [NERvGear/NERvSDK#7]
Here is an overloaded function in string.h

> [NERvSDK/include/NERvGear/string.h](https://github.com/NERvGear/NERvSDK/blob/402e25aa3d16c68e1427d7d1e78e6295fc46d87e/include/NERvGear/string.h)(140)
> 
> <pre><code>/// \overloadfun{NERvCopyString(size_t, const wchar_t*, size_t, wchar_t*)}
> ///
> /// This version is used when you don't know the length of the source string, the string's length will be counted if necessary.
> ///
> /// \header{NERvGear/string.h}
> static inline size_t NERvCopyString(const wchar_t* src, size_t nCharDst, wchar_t* dst)
> {
>     if (dst)
>         return PrivateCopyString(src, nCharDst, dst);
> <br>
>     return NERvStringLength(src);
> }</code></pre>

According to the comment above, this function should return the same value as another overloaded function does under certain condition, which is supposed to be <code>nCharSrc</code>.

> [NERvSDK/include/NERvGear/string.h](https://github.com/NERvGear/NERvSDK/blob/402e25aa3d16c68e1427d7d1e78e6295fc46d87e/include/NERvGear/string.h)(113)
> 
> <pre><code>/// \brief Copies a null-terminated string to the destination buffer.
> ///
> <strong>/// \param [in]  nCharSrc The size of the \a src buffer, in characters, including the null terminator.</strong>
> ...
> /// \return
> ...
> <strong>///     If \a dst is NULL and \a nCharDst is 0, the return value is \a nCharSrc.\n</strong>
> ///     If the function fails, the return value is 0.
> ///
> /// \see
> ///     NERvCopyString(const wchar_t*, size_t, wchar_t*), NERvCopyString(const wchar_t (&)[N_CHAR], size_t, wchar_t*)
> ///
> /// \header{NERvGear/string.h}
> static inline size_t NERvCopyString(size_t nCharSrc, const wchar_t* src, size_t nCharDst, wchar_t* dst)
> {
>     if (dst)
>         return PrivateCopyString(src, nCharDst, dst);
> <br>
>     return nCharSrc;
> }
> </code></pre>

Also, the comment points out that <code>nCharSrc</code> includes the null terminator.
In this case, back to the first overloaded function, it seems that this function is supposed to return the length includes the null terminator.<br>
In conclusion, perhaps the first function should be look like the one below.

<pre><code>static inline size_t NERvCopyString(const wchar_t* src, size_t nCharDst, wchar_t* dst)
{
    if (dst)
        return PrivateCopyString(src, nCharDst, dst);
<br>
<strong>    return NERvStringLength(src) + 1;</strong>
}</code></pre>
